dev/com.ibm.ws.beanvalidation.v20.cdi/src/com/ibm/ws/beanvalidation/v20/cdi/internal/BvalManagedObjectBuilderImpl.java

```diff
@@ -192,10 +249,15 @@ private void createManagedConstraintValidatorFactory(Configuration<?> config, Cl
         BootstrapConfiguration configSource = config.getBootstrapConfiguration();
         String constraintValidatorFactoryClassName = configSource.getConstraintValidatorFactoryClassName();
         ConstraintValidatorFactory cvf = null;
-
         if (constraintValidatorFactoryClassName == null) {
             // use default
-            cvf = createManagedObject(InjectingConstraintValidatorFactory.class);
+            if (isBeanValidationVersion31()) {
```

Comment:
Wouldn't you be better off creating a new version of this class in a different bundle, to have a clean separation between bVal 3.0 (hibernate 7) and bVal 3.1 (hibernate 9)?